File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService",function ($http) {

    this.findAll = function () {
        return $http.post("../brand/findAll.do");
    };

    this.findPage = function (pageNum,pageSize) {
        return $http.get("../brand/findPage.do?pageNum="+pageNum+"&pageSize="+pageSize);
    };

    this.findOne = function (id) {
        return $http.get("../brand/findById.do?id="+id);
    };

    this.save = function (newBrand) {

        return $http.post("../brand/save.do",newBrand);
    };

    this.update = function (newBrand) {
        return $http.post("../brand/update.do",newBrand);
    };

    this.dele = function (checkIds) {
        return $http.get("../brand/deleteByIds.do?ids="+checkIds);
    };

    this.search = function (pageNum, pageSize,searchBrand) {
        return $http.post("../brand/findByCondition.do?pageNum="+pageNum+"&pageSize="+pageSize,searchBrand);
    }

    this.selectOptionList = function () {
        return $http.get("../brand/selectOptionList.do");
    }

});

File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * @author Administrator
 */
public interface BrandService {

	public abstract List<TbBrand> findAll();

	public abstract PageResult findPage(int pageNum, int pageSize);

	public abstract void save(TbBrand brand);

	public abstract void updateById(TbBrand brand);

	public abstract TbBrand findById(Long id);

	public abstract void deleteByIds(Long[] ids);

	public abstract PageResult findByCondition(TbBrand brand, int pageNum, int pageSize);

	public abstract List<Map> selectOptionList();

}
